### PR TITLE
Add ability to enable/disable coupons

### PR DESCRIPTION
**Pull request by syamsoul**
can enable or disable the coupon based on `is_enable` column in `coupons` table

### CODE REVIEW

File: database/migrations/create_coupons_table.php.stub [hunk 1]
```diff
@@ -23,6 +23,7 @@ return new class extends Migration
             $table->unsignedInteger('limit')->nullable();
             $table->nullableMorphs('redeemer');
             $table->timestamp('expires_at')->nullable();
+            $table->boolean('is_enable')->default(true);
```
Comment (michael-rubel) [inline database/migrations/create_coupons_table.php.stub:26]: I'd rename the column from `is_enable` to `is_enabled`.

File: src/Models/Coupon.php [hunk 2]
```diff
@@ -82,6 +83,26 @@ public function isNotExpired(): bool
...
     }
 
+    /**
+     * Check if code is enabled.
+     *
+     * @return bool
+     */
+    public function isEnabled(): bool
+    {
+        return $this->is_enable;
```
Comment (michael-rubel) [inline src/Models/Coupon.php:93]: There should be a `getIsEnabledColumn()` method in the `DefinesColumns` trait to make the column name overrideable.

```suggestion
        return $this->{static::$bindable->getIsEnabledColumn()};
```

**Review by michael-rubel — CHANGES_REQUESTED** [review]

**Comment by michael-rubel** [comment]
Hey there @syamsoul 👋

Please, fix PR checks and review my suggestions.

**Note**: BC check is probably acceptable since this is the model's `casts` and I don't see the need to rely on the default value of this property.

**Comment by syamsoul** [comment]
> Hey there @syamsoul 👋
> 
> Please, fix PR checks and review my suggestions.
> 
> **Note**: BC check is probably acceptable since this is the model's `casts` and I don't see the need to rely on the default value of this property.

hi @michael-rubel , i just did some amendments based on your request.

**Comment by michael-rubel** [comment]
@syamsoul tests are still failing.

Also, I'd add the default value `true` for the `isEnabled` method for those who still don't have the `is_enabled` column in their database.

**Comment by syamsoul** [comment]
> @syamsoul tests are still failing in PR.
> 
> Also, I'd add the default fallback to `true` in the `isEnabled` method for those who still don't have the `is_enabled` column in their database.

@michael-rubel , I just pushed new commits but not sure why 3 tests are still failing

**Comment by michael-rubel** [comment]
@syamsoul I'll take a look.

**Review by michael-rubel — APPROVED** [review]

**Comment by michael-rubel** [comment]
Thank you @syamsoul 

Released as [v7.1.0](https://github.com/michael-rubel/laravel-couponables/releases/tag/7.1.0).

**Comment by syamsoul** [comment]
> Thank you @syamsoul
> 
> Released as [v7.1.0](https://github.com/michael-rubel/laravel-couponables/releases/tag/7.1.0).

Thank you very much !

**Pull request merged**